### PR TITLE
The Friendship Update

### LINKED ISSUE #14: Create Friend Invite
We want users to be able to email their friends if they would like them to join in on the MyWeek fun! I will work on this logic, as well as general emails if someone would like to just inform their 92-year-old aunt of the event!

### LINKED ISSUE #17: SMS Function/Reminders
Integrate an SMS function that will allow the user to get a text message before their event begins.

### LINKED ISSUE #19: User Profile & Edit
The user profile page will allow users to see their information, edit information, and show their friends list.

### PR DESCRIPTION
**Pull request by ewink1977**
Update details:

- This update will close #14 as on the event detail page, users will be able to email an off-platform user and invite them to join MyWeek. Future functionality could have the user automatically be added to the event upon registering.
- This update will close #19 as a User Profile has been created, with the ability of the owner of that profile to edit their details. The user profile lists all PUBLIC events the user has created. I still need to add a list of events the user is invited to but did not create. 
- This update can close #17 as SMS functionality has been incorporated and is working. Due to the complexity of making 'scheduled' SMS messages, right now the function only notifies you when you have been invited to an event. This function does not kick in if the user doesn't have a phone number attached to their account.

**NOTE - NEITHER THE EMAIL OR SMS FUNCTIONALITY WILL WORK WITHOUT THE .ENV FILE AND WITHOUT PYTHON-DOTENV BEING INSTALLED IN YOUR VENV.** I will email you all the file, but you **MUST MAKE SURE** `.env` is in your `.gitignore` list. The SMS functionality will also need the `twilio` library installed as well (my piplock file will reference it, I forgot to update the requirements.txt file).

### CODE REVIEW

**Comment by nheyland** [comment]
Do you want to throw the Django and Mapbox ones in there too while you're making it? just for good practice even if its a little late for those

**Comment by ewink1977** [comment]
> Do you want to throw the Django and Mapbox ones in there too while you're making it? just for good practice even if its a little late for those

I already have, actually, it's just a matter of changing the settings.py and other py files. It's a bit more complicated to hide it in JavaScript, so that may be something we need a JS expert to help us with.

File: planner/templates/details.html [hunk 1]
```diff
@@ -40,6 +63,39 @@ <h4>{{event.address}}</h4>
         });
 
     </script>
+    <!-- YOU DON'T GET TO ADD A FRIEND TO THIS IF YOU'RE NOT THE PERSON WHO CREATED THE EVENT. -->
+    {% if event.created_by.id == request.session.user_id %}
+    <!-- PLACING SOCIAL PIECES OF THE PLANNER DOWN HERE FOR NOW. FRONT-END/LEAD CAN MOVE TO WHEREEVER. /ew -->
+    <h2>ADD A FRIEND PLACEHOLDER</h2>
```
Comment (ewink1977) [inline planner/templates/details.html:69]: Whoops. This should say something better. Can be easily changed.

**Comment by nheyland** [comment]
Is this ready for final review and merge?

**Comment by ewink1977** [comment]
> Is this ready for final review and merge?

Yes. I think you can ignore or delete all the files from this PR that are in conflict, though you should probably make the .gitignore in this pull request the working one so that my .env file doesn't get pushed up by anyone on accident.

**Comment by ewink1977** [comment]
And I'm an idiot and accidentally closed this. Reopening.

**Comment by nheyland** [comment]
Roger that, I was trying to get my gitignore to hit the repo but could not do it for some reason, Ill merge this today as well thanks @ewink1977 !

File: .gitignore [hunk 2]
```diff
@@ -7,10 +7,12 @@
...
-# db.sqlite3-journal
+db.sqlite3
+db.sqlite3-journal
 media
 .vscode
+login\__pycache__
+
 
 # If your build process includes running collectstatic, then you probably don't need or want to include staticfiles/
 # in your Git repository. Update and uncomment the following line accordingly.
```
Comment (nheyland) [inline .gitignore:21]: these are showing up in conflicts, Im not 100% but I think adding login/__pycache__/* to .gitignore and same for other apps will prevent the git from bringing it it. 

login/__pycache__/__init__.cpython-39.pyc
login/__pycache__/models.cpython-39.pyc
login/__pycache__/render.cpython-39.pyc
login/__pycache__/urls.cpython-39.pyc
login/__pycache__/views.cpython-39.pyc
myWeek/__pycache__/__init__.cpython-39.pyc
myWeek/__pycache__/settings.cpython-39.pyc
myWeek/__pycache__/urls.cpython-39.pyc
myWeek/__pycache__/wsgi.cpython-39.pyc
planner/__pycache__/__init__.cpython-39.pyc
planner/__pycache__/admin.cpython-39.pyc
planner/__pycache__/geocode.cpython-39.pyc
planner/__pycache__/models.cpython-39.pyc
planner/__pycache__/urls.cpython-39.pyc
planner/__pycache__/utils.cpython-39.pyc
planner/__pycache__/views.cpython-39.pyc

**Pull request merged**